### ISSUE TITLE
[Wayland KDE] Flameshot doesn't capture full desktop w/ multi-monitor

### ISSUE THREAD

**Issue by Riven-Spell** [flameshot-org/flameshot#3073]
### Flameshot Version

Flameshot v12.1.0
Compiled with Qt 5.15.5

### Installation Type

User repository (AUR)

### Operating System type and version

Arch Linux (Linux aetherflow 6.1.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 18 Jan 2023 19:54:38 +0000 x86_64 GNU/Linux)

### Description

Flameshot seems to not realize my second monitor exists, while still taking a picture of it. Ctrl+A'ing the screenshot only gets what's shown on my ultrawide.

My setup is a 16:9 1080p monitor on the left-hand side of a 21:9 1080p monitor.

Moving the left monitor to be right of the ultrawide in software allows me to be able to take a proper screenshot on the ultrawide, but still the length is limited to just the ultrawide.

Setting the other monitor as the primary does not affect the position of flameshot or the size of pictures it takes; it just continues to display on the ultrawide.

### Steps to reproduce

1. Run `flameshot gui` on a multi-monitor KDE Wayland setup

Profit.

### Screenshots or screen recordings

![Screenshot_20230126_035353](https://user-images.githubusercontent.com/13892402/214936752-7cbb1397-dbbf-4c28-8ada-69ff43efb4f0.png)

### System Information

1. Arch Linux (Linux aetherflow 6.1.7-arch1-1 #1 SMP PREEMPT_DYNAMIC Wed, 18 Jan 2023 19:54:38 +0000 x86_64 GNU/Linux)
2. ![Screenshot_20230126_040020](https://user-images.githubusercontent.com/13892402/214938173-3c55524c-29d4-4e68-b903-f2ee2876eefd.png)
3. KDE/kwin/Wayland

**Comment by Chryseus** [comment]
Can confirm I'm having the exact same issue here, also running Arch with Zen kernel, KDE and Wayland.
Monitor setup:

![image](https://user-images.githubusercontent.com/44627250/214971628-a253e080-562b-4968-95f5-e1507aac29ab.png)

**Comment by UVClay** [comment]
Same behavior on Fedora.  Flameshot also takes significantly longer to launch on Wayland than it does on X.  When pressing my shortcut on X, Flameshot triggers almost immediately.  When running Wayland, it takes between 3 and 7 seconds to start.  Spectacle seems not to have any issues with this (but I prefer using Flameshot).

![Flameshot](https://user-images.githubusercontent.com/295806/215285348-df5e442f-bb1e-42d5-a0ef-e8dd19b1970a.png)
![Screenshot_20230128_134407](https://user-images.githubusercontent.com/295806/215285351-0ff855f7-5119-4daa-be89-ca4f82d97163.png)
![Screenshot_20230128_134554](https://user-images.githubusercontent.com/295806/215285382-1a432298-18d3-4464-8f3f-9f1ba7d5bc04.png)

**Comment by meirg** [comment]
I believe this is a duplicate of #2364

**Comment by Combinacijus** [comment]
Similar problem with 2 external monitors. Green good, yellow black bar, red not in capture area. 
I think it happened after updating from Windows 10 to Windows 11 although it might happened later after update I'm not sure, noticed just today, maybe some update messed it up
V12.1.0, V12.0
![image](https://github.com/flameshot-org/flameshot/assets/16375702/a1c47a7e-804a-4962-a14d-0b0ee29039ee)

**Comment by r3tr0g4m3r** [comment]
This problem seems to be investigated in [#2364](https://github.com/flameshot-org/flameshot/issues/2364), i work my way around it launching flameshot from flatpak with the environment variable "QT_QPA_PLATFORM=xcb", to set it i use Flatseal and configure it in "Environment", this however causes the keyboard to not function, only mouse control.

**Comment by moosyeen** [comment]
I also have a familiar issue on KDE plasma wayland,  I can take a screenshot on the primary screen while there are two exist. Besides, I can only save the picture but can't copy it to my clipper board. 

Version information is:
```
Flameshot v12.1.0 ()
Compiled with Qt 5.11.3
linux: 5.19.0-46-generic
ubuntu: 22.04
```

**Comment by Seraphli** [comment]
Same with KDE plasma. Multiple monitors and one of the monitors is rotated.

**Comment by Lightfire228** [comment]
I was able to work around this issue by forcing the window geometry
![image](https://github.com/flameshot-org/flameshot/assets/15241322/fa03497a-c6d3-42e0-b6e9-6fe18830579f)

**Comment by StevenGee3398** [comment]
Thanks a lot @Lightfire228 this works perfectly :)

**Comment by Seraphli** [comment]
Restart flameshot works for me.

**Comment by miachm** [comment]
@Lightfire228 worked for me in Archlinux. Although "Ignore requested geometry" has been renamed to "Obey geometry restrictions". So you have to mark the opposite option:

![image](https://github.com/flameshot-org/flameshot/assets/10280072/44027a75-faf5-4fc4-a916-5b5b7fd47510)

**Comment by maxneuds** [comment]
Just another confirmation. Window rules perfectly fixes it. Thank you @Lightfire228

**Comment by Antonio-P8A** [comment]
> I was able to work around this issue by forcing the window geometry
> 
> ![image](https://private-user-images.githubusercontent.com/15241322/271452128-07fc0612-edab-4c34-94ef-8abfc60c23c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQzMjEyMTQsIm5iZiI6MTcxNDMyMDkxNCwicGF0aCI6Ii8xNTI0MTMyMi8yNzE0NTIxMjgtMDdmYzA2MTItZWRhYi00YzM0LTk0ZWYtOGFiZmM2MGMyM2MwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA0MjglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNDI4VDE2MTUxNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWIwNGQyNTYyM2U1OGEzZTkwMDI4Y2FiMGI4NDUwYjQ3NTY0OGJiZGVjNGI2M2QxYzU2ZDc4OWZlNWQwN2E5NzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.e6hGwUM619neKFvc8um5TwC_89cBwADK2x_bJhi_OqI)
> 
> [Source thread](https://github.com/flameshot-org/flameshot/issues/2364#issuecomment-1689111940)

Funcionó perfectamente, muchas gracis

**Comment by luisantoniojr** [comment]
Just for you to know, maybe it can help.
I have a laptop and one monitor plugged in, on laptop the resolution is QHD and on the monitor it is HD.
It only works when i set the same resolution for both monitors, in this case both to 1920x1080, than the Window rules fixes the issue.
Thank you @Lightfire228.

**Comment by mingue** [comment]
I have also the same issue, on plasma 6 with Wayland, the fix will only work if I have the same resolution in both monitors as @luisantoniojr mentioned, but also, I need to have the same scaling for both monitors, if one is 100% and the other one 110%, it won't work either.

Thanks!

**Comment by DuncanMcRae** [comment]
The workaround works for Arch Wayland KDE6 using laptop and two additional monitors. And its a non-starter as my laptop screen is not 16:9.

**Comment by Riven-Spell** [comment]
> I was able to work around this issue by forcing the window geometry
> 
> ![image](https://private-user-images.githubusercontent.com/15241322/271452128-07fc0612-edab-4c34-94ef-8abfc60c23c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTYwNzA1NDUsIm5iZiI6MTcxNjA3MDI0NSwicGF0aCI6Ii8xNTI0MTMyMi8yNzE0NTIxMjgtMDdmYzA2MTItZWRhYi00YzM0LTk0ZWYtOGFiZmM2MGMyM2MwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MTglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTE4VDIyMTA0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTk4MDRiOGFkZWJjMzRlOGMwZDI2OGJlMjUzY2JkOGQxZjc5ZTE0ZGNhMGQ5MDhmMDI2N2RlN2E3OGNjNDViMjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cTRUHw8g57hPqGwL33k4lOuT82QpAJJbGgraRa6h6fk)
> 
> [Source thread](https://github.com/flameshot-org/flameshot/issues/2364#issuecomment-1689111940)

This is great, thanks a million. Just coming back to try daily driving Linux again.

**Comment by nddragoon** [comment]
The workaround above doesn't seem to work for me. It just sets the top left corner of the flameshot window to the center of my primary display. Changing the position values doesn't seem to do anything.

**Comment by corincorvus** [comment]
It not work for me. 
Standard:
![PXL_20240718_110856497](https://github.com/user-attachments/assets/401b50fe-5057-456e-a7d4-06fb81d498fb)


If i change to the Workaround Flameshot moves to right bot corner.
![IMG_20240718_130838](https://github.com/user-attachments/assets/668d338d-3abc-49c5-912a-9a42f4b4d2d9)

**Comment by ThePrivacyPlayer** [comment]
How is this bug still listed as unconfirmed?

I love flameshot, been using it for a while on Mac for my work, but I was going crazy about this and almost went to search for another app. Using Nobara KDE Wayland.

Lucky that I found this thread, Window rules saved my day :-)

Still, this shouldn't be listed as unconfirmed anymore given the amount of people that have confirmed it on this thread, including myself.

@corincorvus Have you tried using the exact settings from miachm's reply? (https://github.com/flameshot-org/flameshot/issues/3073#issuecomment-1985512253). From what I can see on your screenshots, that's not what you're using, and this fixed it for me (it didn't work before that)

**Comment by tonyd33** [comment]
Just in case someone else is stumbling upon this thread while trying to fix this on Hyprland like I did, the following rules fixed it for me:
```
# noanim isn't necessary but animations with these rules might not look good. use at your own discretion.
windowrulev2 = noanim, class:^(flameshot)$
windowrulev2 = float, class:^(flameshot)$
windowrulev2 = move 0 0, class:^(flameshot)$
windowrulev2 = pin, class:^(flameshot)$
```

**Comment by corincorvus** [comment]
> How is this bug still listed as unconfirmed?
> 
> I love flameshot, been using it for a while on Mac for my work, but I was going crazy about this and almost went to search for another app. Using Nobara KDE Wayland.
> 
> Lucky that I found this thread, Window rules saved my day :-)
> 
> Still, this shouldn't be listed as unconfirmed anymore given the amount of people that have confirmed it on this thread, including myself.
> 
> @corincorvus Have you tried using the exact settings from miachm's reply? ([#3073 (comment)](https://github.com/flameshot-org/flameshot/issues/3073#issuecomment-1985512253)). From what I can see on your screenshots, that's not what you're using, and this fixed it for me (it didn't work before that)

Yes i tried, but it didnt work. After switch to Tuxedo Linux it work. On Nobara it doesnt work (Tested today)

**Comment by Ubspy** [comment]
I'm having this issue also, running Arch with KDE on Wayland. I added the window rule as [shown here](https://github.com/flameshot-org/flameshot/issues/3073#issuecomment-1985512253). This window rule doesn't work for me sadly, and I am not sure how to fix that workaround.

**Comment by ThePrivacyPlayer** [comment]
The fix from @miachm doesn't appear to work anymore. It was definitely working in v6.10.3 for me, but now in 6.11.2, the issue is there again... (Didn't test in 6.11.0 and 6.11.1)

This needs to be resolved asap!

**Comment by lucemans** [comment]
+1

**Comment by typecasto** [comment]
Also having this issue (how is it still unconfirmed? there's 24 people in this thread and it's been open for almost 2 years...), the workaround works for me, but isn't ideal because I use the pinning feature quite a lot and this also forces pins to the upper left. Giving the pin window it's own class would fix that, at least.

**Comment by GingerBobRoss** [comment]
I am also having this issue as well
FlameShot use to work phenomenal and now its pretty much unusable on primary monitors since it wants to cause the left monitor to be on the primary.
![image](https://github.com/user-attachments/assets/d3670d37-bfdb-4c2d-b3dd-6382b74ca409)

What flameshot captures:
![image](https://github.com/user-attachments/assets/ea260a0e-5951-48f9-ba40-c94b989acd9d)

The screen space it doesnt capture in yellow ( which is A LOT)
![image](https://github.com/user-attachments/assets/38b614c6-0e1d-4cbe-ae8c-54952e27277c)

**Comment by cycloarcane** [comment]
> I was able to work around this issue by forcing the window geometry
> 
> ![image](https://private-user-images.githubusercontent.com/15241322/271452128-07fc0612-edab-4c34-94ef-8abfc60c23c0.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU0ODg4MjgsIm5iZiI6MTczNTQ4ODUyOCwicGF0aCI6Ii8xNTI0MTMyMi8yNzE0NTIxMjgtMDdmYzA2MTItZWRhYi00YzM0LTk0ZWYtOGFiZmM2MGMyM2MwLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI5VDE2MDg0OFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE0NmIzYzViN2U2ZDM1MjQ4OGFiYTAxZWZlNjRhZWQzZGViODNiOWJhYTgzNjIyY2NlMDRhYTFmNmJhMjIyNzAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.sjeYHNJ27kT4McpDVWDDB5nH5mBrYcKhrkpZYHNR5qk)
> 
> [Source thread](https://github.com/flameshot-org/flameshot/issues/2364#issuecomment-1689111940)

This is working great in KDE 6.2.4 (plasma) on Arch Linux. Tysm!

**Comment by smooth-baking-powder** [comment]
> It not work for me. Standard: ![PXL_20240718_110856497](https://private-user-images.githubusercontent.com/21257329/349932550-401b50fe-5057-456e-a7d4-06fb81d498fb.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1MDI4MzEsIm5iZiI6MTczNTUwMjUzMSwicGF0aCI6Ii8yMTI1NzMyOS8zNDk5MzI1NTAtNDAxYjUwZmUtNTA1Ny00NTZlLWE3ZDQtMDZmYjgxZDQ5OGZiLmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI5VDIwMDIxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWUzZGNhYmRhMjdmNTZhNDYyMTU0MWUxMmUyMDA0NzgzNTU2MjFhZWYwMWFjZWJiNzBhYzkzNGRjZWZjODg2OTEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.6KSf0stSu8EPTTrIR1SeeZvSKZ794V3_8oXkzjEQKeY)
> 
> If i change to the Workaround Flameshot moves to right bot corner. ![IMG_20240718_130838](https://private-user-images.githubusercontent.com/21257329/349932238-668d338d-3abc-49c5-912a-9a42f4b4d2d9.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzU1MDI4MzEsIm5iZiI6MTczNTUwMjUzMSwicGF0aCI6Ii8yMTI1NzMyOS8zNDk5MzIyMzgtNjY4ZDMzOGQtM2FiYy00OWM1LTkxMmEtOWE0MmY0YjRkMmQ5LmpwZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDEyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMjI5VDIwMDIxMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTRlNTAzZjkzMjEwNDA1Yzc1ZDI3MzI5YjQ1MWZkMGQxMjhmODEzNjQ5YjU2YWQwYmJjOTMyMDY4ODM0MjQxMWQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.VBF_QZu2_5mMPKsJKynXF8oCzTEVy8bGtvlkAH8WTw4)
> 
> It only work, if i move my above monitore to the left to 0,0. this is crap :D I need a better Solution.. i played with some other options, but nothing worked.
> 
> Nobara Linux KDE Wayland.
> 
> Thanks

With my Setup i now got a Solution with the Flathub Version. The native Installation not work with this Settings. Dont ask me why.
![2024-12-29_21-10](https://github.com/user-attachments/assets/90c64b6e-46ec-4b6b-8948-d6e74893c67f)

**Comment by mactan-sc** [comment]
@smooth-baking-powder tysm working for me on arch kde plasma 6.2.5-1 flameshot 12.1

i wish this program just worked on kde without such a long lived smorgasbord of hacks and workarounds.  some forum posts i  read were over 6 years old

**Comment by Ubspy** [comment]
How do we go about getting this bug confirmed? There seem to be enough reports that this isn't just a 1 off misconfiguration.

**Comment by SuperVivid** [comment]
I confirm the same Problems like others here told. Arch, Wayland, KDE. The Workaround with the Window Rules does not work for me.

**Comment by MfCrizz** [comment]
I conform that I have this bug on kubuntu 24.04 lts with Wayland too. The Workaround with the Window Rules does not work for me.

**Comment by RunsOnRum** [comment]
> [@Lightfire228](https://github.com/Lightfire228) worked for me in Archlinux. Although "Ignore requested geometry" has been renamed to "Obey geometry restrictions". So you have to mark the opposite option:
> 
> ![image](https://github.com/flameshot-org/flameshot/assets/10280072/44027a75-faf5-4fc4-a916-5b5b7fd47510)

Seems to be there for me. With added note

![Image](https://github.com/user-attachments/assets/8c5b2185-6abf-4265-b853-92e3850c5034)

**Comment by ArjixWasTaken** [comment]
> This problem seems to be investigated in [#2364](https://github.com/flameshot-org/flameshot/issues/2364), i work my way around it launching flameshot from flatpak with the environment variable "QT_QPA_PLATFORM=xcb", to set it i use Flatseal and configure it in "Environment", this however causes the keyboard to not function, only mouse control.

I installed the flatpak, added the environment variable and it works perfectly fine!
the solution of "ignore requested geometry" did not work for me, I am on Arch

**Comment by ayylmaonade** [comment]
For anybody having issues with the recent KDE updates -- here's a solution. Just add the size parameter to your window rule, and multiply the horizontal resolution of your monitors while keeping the same vertical resolution size. So in my example, I have 2 2560x1440p monitors. 2560*2 = 5120. Easy.

<img width="1208" height="914" alt="Image" src="https://github.com/user-attachments/assets/019c97c3-9d60-4a8b-b266-be79a70986cb" />

**Comment by vachan-t** [comment]
@ayylmaonade tysm

**Comment by vachan-t** [comment]
since my screens are different size going from the smaller one to the bigger one can introduce transparent parts into the screenshot, not a problem just something that happens and is probably use full sometimes

**Comment by murolem** [comment]
Same Issue, Kubuntu KDE Plasma. 

> I was able to work around this issue by forcing the window geometry
> 
> ![image](https://github.com/flameshot-org/flameshot/assets/15241322/07fc0612-edab-4c34-94ef-8abfc60c23c0)
> 
> [Source thread](https://github.com/flameshot-org/flameshot/issues/2364#issuecomment-1689111940)

Sadly didn't help, it still appears scuffed but in another way.

**Comment by Ubspy** [comment]
> For anybody having issues with the recent KDE updates -- here's a solution. Just add the size parameter to your window rule, and multiply the horizontal resolution of your monitors while keeping the same vertical resolution size. So in my example, I have 2 2560x1440p monitors. 2560*2 = 5120. Easy.
> <img alt="Image" width="1208" height="914" src="https://private-user-images.githubusercontent.com/3524987/474634868-019c97c3-9d60-4a8b-b266-be79a70986cb.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTQ4NzgwMzUsIm5iZiI6MTc1NDg3NzczNSwicGF0aCI6Ii8zNTI0OTg3LzQ3NDYzNDg2OC0wMTljOTdjMy05ZDYwLTRhOGItYjI2Ni1iZTc5YTcwOTg2Y2IucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDgxMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA4MTFUMDIwMjE1WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZDA2N2E5ZGJmOTA3NTIyMTZmMDZhMWM2MTg3NmY2ZjZiYjRlYmE4MjE3MTI0MjcxNGNjZjgwNDM0NDFhMDJiZCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.7UZzeOXToV539Du5AcA02VLjCQyAbWYz1FL7o7zmep8">

This worked great, at least I have this until it's properly fixed.

**Comment by Hunanbean-Collective** [comment]
Had the same issue.

**Comment by Dylan-86** [comment]
Same issue here, Ubuntu 24.04 + KDE Plasma on Wayland

**Comment by GabubuAvailable** [comment]
Heya! I come with an update and a workaround until they will fix it on the next update...
So, apparently what @ayylmaonade did on https://github.com/flameshot-org/flameshot/issues/3073#issuecomment-3155909580 is the perfect option!

However, if you find the issue persisting, you need to go back to "Window Behaviour", then "Advanced" tab. Once you are there, make sure you change the "Window placement" to "Random".
<img width="1248" height="776" alt="Image" src="https://github.com/user-attachments/assets/aeac071c-a6ed-4baf-82ec-e98d3a6c3617" />

So in short, you need to do both the window rule ayylmaonade made _(exactly as it is, except the size one where you need to do a math)_ + changing the window placement to random.

Hope this helps! ^^
Gabubu

**Comment by disasmwinnie** [comment]
> This problem seems to be investigated in [#2364](https://github.com/flameshot-org/flameshot/issues/2364), i work my way around it launching flameshot from flatpak with the environment variable "QT_QPA_PLATFORM=xcb", to set it i use Flatseal and configure it in "Environment", this however causes the keyboard to not function, only mouse control.

Having the same issue on Gnome 48 (Wayland) on Arch Linux. The workaround _QT_QPA_PLATFORM_ helped.

```console
flameshot --version
    Flameshot v13.1.0 ()
    Compiled with Qt 6.9.1
```

**Comment by huyhoang8398** [comment]
following

**Comment by FlanGrande** [comment]
> Heya! I come with an update and a workaround until they will fix it on the next update... So, apparently what [@ayylmaonade](https://github.com/ayylmaonade) did on [#3073 (comment)](https://github.com/flameshot-org/flameshot/issues/3073#issuecomment-3155909580) is the perfect option!
> 
> However, if you find the issue persisting, you need to go back to "Window Behaviour", then "Advanced" tab. Once you are there, make sure you change the "Window placement" to "Random". <img alt="Image" width="1248" height="776" src="https://private-user-images.githubusercontent.com/88589756/486518175-aeac071c-a6ed-4baf-82ec-e98d3a6c3617.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTkyMjczOTMsIm5iZiI6MTc1OTIyNzA5MywicGF0aCI6Ii84ODU4OTc1Ni80ODY1MTgxNzUtYWVhYzA3MWMtYTZlZC00YmFmLTgyZWMtZTk4ZDNhNmMzNjE3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA5MzAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwOTMwVDEwMTEzM1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZjZmJiZDNjODU3ZjlhZjI0NzZmYjlhNGFjMTNmMTkyNTk1NWU1YjYwYjMxYjRmYjIxMmI2NzkyM2FkMDQ4NjYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Qd0-xT6dkLEzdacRKkKT_I-dDemVBKUiERH1hs6aLKA">
> 
> So in short, you need to do both the window rule ayylmaonade made _(exactly as it is, except the size one where you need to do a math)_ + changing the window placement to random.
> 
> Hope this helps! ^^ Gabubu

Gabubu you are MVP. Exactly the rules and this last option is what did it for me.
For reference, I have left screen 1080p, right screen 1440p, both centered at the same height (so there's some "space" above and below the 1080p screen)

I can't comprehend either how this is Unconfirmed Bug!!!

**Comment by FlanGrande** [comment]
@GabubuAvailable I just realized, you can also add Random as a specific part of the rules, so it doesn't need to affect all windows, only Flameshot!

<img width="2036" height="63" alt="Image" src="https://github.com/user-attachments/assets/1a14faa5-a882-434b-8dad-301254c47213" />

**Comment by RasT110e5** [comment]
Confirms this workaround works in Debian KDE Plasma

**Comment by DuendeInexistente** [comment]
Can confirm forcing the properties still works in kde 6.5.5, thankee

**Comment by FlanGrande** [comment]
Having the workaround is fine, and it has worked well so far for me. Surely though, some actual fix should be implemented on Flameshot to have it work with several screens without the need for the workaround?

I know it's being talked about in #2364 however most answers there seem to be talking about a separate workaround, usually related to setting QT_QPA_PLATFORM=xcb

I encourage the team to find an actual solution and maybe create a separate thread or move them together, but in any case try to steer the conversation towards solutions rather than workarounds.

**Comment by Ketrel** [comment]
Forcing the properties isn't helping for me.

Flameshot: v13.3.0  
Plasma: 6.6.5  
Distro: Arch  
Backend: wayland

<img width="939" height="680" alt="Image" src="https://github.com/user-attachments/assets/e1b81327-a5db-4e84-b3db-eca85c5a2209" />

I'm not sure if this is the cause or not, but note I have the second monitor set as primary.

`[Monitor-0] [Monitor-1]`  with Monitor-1 being primary

When I issue `flameshot gui`, the UI shows up on Monitor-0 only, and only allows capture from Monitor-0.  Clicking anywhere on Monitor-1 causes flameshot to lose focus.

I tried with the windows rules mentioned above, but all that accomplishes is the flameshot gui appears on Monitor-1, but shows the contents of Monitor-0 for selecting.

**Comment by mrjones-plip** [comment]
@Ketrel et al. - 14.0.rc3 was [released](https://github.com/flameshot-org/flameshot/releases/tag/v14.0.rc3) 4 days ago. Maybe give that a try? GA should be soon!

> This is hopefully the final beta before the v14 release.

**Comment by mmahmoudian** [comment]
I'm pretty sure this is fixed in v14. As @mrjones-plip kindly pointed out,third and perhaps the last beta version va v14 was released 4 days ago.

For this reason, I'm going to close this issue. If your still was facing this issue in v14 or later, please comment here or opena new issue.

**Comment by Nv7z2** [comment]
(for flameshot version 13.3.x, no tweaks needed for flameshot 4.0.0-1+)

If anyone still got this problem on KDE Plasma - I found not optimal, but working workaround by settings "Size" to 3840x1080 in Window Rules. 3840px because I use 2x full-hd monitors side by side. The screenshot is in Polish language, but I guess you can read it based on icons.

<img width="1183" height="828" alt="Image" src="https://github.com/user-attachments/assets/8a868000-313a-4f8c-94f1-9be89ce3b8fe" />

<img width="905" height="831" alt="Image" src="https://github.com/user-attachments/assets/c9e5e073-ffd4-400d-a95a-ac5eced03d99" />
